core/src/main/scala/org/http4s/internal/parsing/Rfc7235.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2013-2020 http4s.org
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.http4s.internal.parsing
+
+import cats.syntax.all._
+import cats.data.NonEmptyList
+import cats.parse.Parser.{char, charIn}
+import cats.parse.{Parser, Parser1}
+import cats.parse.Rfc5234.{alpha, digit, sp}
+import org.http4s.{Challenge, Credentials}
+import org.http4s.internal.parsing.Rfc7230.{bws, headerRep1, ows, quotedString, token}
+import org.http4s.util.CaseInsensitiveString
+
+private[http4s] object Rfc7235 {
+  /*  token68 = 1*( ALPHA / DIGIT / "-" / "." / "_" / "~" / "+" / "/" )
+   *  *"="
+   * */
+  val t68Chars: Parser1[Char] = charIn("-._~+/").orElse1(digit).orElse1(alpha)
+
+  val token68: Parser1[String] = (t68Chars.rep1 ~ charIn('=').rep).string
+
+  val authParamValue: Parser[String] = token.orElse1(quotedString)
+  // auth-param = token BWS "=" BWS ( token / quoted-string )
+  val authParam: Parser1[(String, String)] =
+    (token <* (bws ~ char('=').void ~ bws)) ~ authParamValue
+
+  /*
+    challenge = auth-scheme [ 1*SP ( token68 / [ ( "," / auth-param ) *(
+    OWS "," [ OWS auth-param ] ) ] ) ]
+   */
+  val challenge: Parser1[Challenge] =
+    ((token <* sp) ~ Parser.repSep(authParam.backtrack, 0, ows <* char(',') *> ows).map(_.toMap))
```

Comment:
`Rfc7230.headerRep1` should also take care of the repetition of the`authParam`.  It's an expansion of the `#` rule in the HTTP RFCs.

Reply:
Doesn't the token68 need to be `orElse1` with the auth-params here?  I'm not following how this parses basic auth.

Reply:
Challenge.authScheme should probably also be CaseInsensitiveString for consistency with Credentials, but we cant do that in 0.21.

Reply:
we dont really have a type to deal with token68 here. How should that be interpreted? This should be added for 1.0, since we cant really change api for 0.21.

Credentials are handles elsewhere.

Reply:
> `Rfc7230.headerRep1` should also take care of the repetition of the`authParam`. It's an expansion of the `#` rule in the HTTP RFCs.

My interpretation here is that there might be no parameters. headerRep1 requires at least one.

Reply:
Oops, you're right.  Well, there could be a `headerRep`.  I think this won't be an unheard of pattern.  It's common enough that RFC7230 extends the standard grammar for it, anyway.

Reply:
Oh, I see.  Our model for `Challenge` doesn't support the token68 form.  Yeah, I think we're kind of stuck.  This one is probably going to require some follow-ups in 1.0 when we can correct model issues.